leader/leader.go

```diff
@@ -105,15 +105,17 @@ func (l *Client) AddWatchCallback(key string, retry time.Duration, fn CallbackFn
 	}()
 }
 
-func (l *Client) getWatchAtLatestIndex(ctx context.Context, api client.KeysAPI, key string, retry time.Duration) (client.Watcher, *client.Response, error) {
-	resp, err := l.getFirstValue(key, retry)
+func (l *Client) recreateWatchAtLatestIndex(ctx context.Context, api client.KeysAPI, key string, entry *log.Entry) (client.Watcher, *client.Response, error) {
+	resp, err := api.Get(ctx, key, nil)
 	if err != nil {
-		return nil, nil, trace.BadParameter("%v unexpected error: %v", ctx.Value("prefix"), err)
-	} else if resp == nil {
-		log.Debugf("%v client is closing, return", ctx.Value("prefix"))
+		return nil, nil, trace.Wrap(err, "failed to fetch value")
+	}
+	if resp == nil {
+		entry.Debug("client is closing")
 		return nil, nil, nil
 	}
-	log.Debugf("%v got current value '%v' for key '%v'", ctx.Value("prefix"), resp.Node.Value, key)
+
+	entry.Debugf("got current value %q for key %q", resp.Node.Value, key)
```

Comment:
this is super verbose to be honest, and our logs are filled with it, can we remove this line